### DISCUSSION TITLE
How do I specify a root for the standalone CLI?

### DISCUSSION THREAD

**Discussion by einkleindatagal** [tailwindlabs/tailwindcss#14698] [Help] [answered]
Hello,

I apologize if I've missed an existing answer, I'm just getting started with Tailwind and Flask, and all of the samples and examples I've seen make it look very simple. And after a day of hacking at this I have no idea what I've missed.

I wish that the Standalone CLI was [part of the documentation](https://tailwindcss.com/docs/installation) instead of a [separate page](https://tailwindcss.com/blog/standalone-cli).

I'm working in a linux devcontainer with VSCode, and initially set up the standalone CLI to run out of /opt/tailwindcss with the latest 3.4.13 binary and a symlink to make it easy to refer to just tailwindcss anywhere. 

I set up my flask application with an app/static folder containing two sub folders, one for src/main.css, and one for dist/css/style.css, and an app/templates folder with index.html like

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>app</title>
    <link rel="stylesheet" href="{{ url_for('static', filename='dist/css/style.css') }}">
</head>
<body class="bg-red-700">
    <h1 class="text-blue-600">Hello World!</h1>
</body>
</html>
```
My tailwind.config.js lives at src/tailwind.config.js, and then my code is src/app/*. The file contains

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './app/templates/**/*.html',
    '/workspaces/project/src/app/templates/index.html',
    '../../workspaces/project/src/app/templates/**.*.{html,js}'
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

When I run my linked and aliased tailwindcss -i ./src/main.css -o ./dist/css/style.css --watch, it generates a file, but doesn't detect any of the classes specified. I've also tried moving the binary to a couple locations within my project assuming maybe the location of it would be treated as the root, but it doesn't seem to make a difference. 

Per

>Paths are relative to your project root, not your tailwind.config.js file, so if your tailwind.config.js file is in a custom location, you should still write your paths relative to the root of your project.

on https://tailwindcss.com/docs/content-configuration#configuring-source-paths, is the tailwind.config.js not the root for the standalone CLI either?

How do I specify a standalone CLI root that works? Am I missing something obvious? This feels like it should be a slam dunk in a few minutes, not a day's worth of trial, error, and research. 

A couple guides I referenced:
* https://testdriven.io/blog/flask-htmx-tailwind/
* https://tute.io/install-configure-tailwind-css-standalone-cli

**Comment by adamwathan** [comment]
Try using the relative path option to make your content paths relative to your `tailwind.config.js` file to see if that gets things working:

https://tailwindcss.com/docs/content-configuration#using-relative-paths

What you're running into doesn't sound like a standalone CLI specific issue, more just an issue because of an unfortunate decision we made years ago about how those paths work by default in the config file. It'll all be relative to the config file by default in v4 thank god.

**Reply by einkleindatagal** [reply, marked as answer]
Once I started using the config file as the root location for everything I was able to get it to work, but that's directly contradicting what the documentation says. When you launch the new version, I hope that the standalone CLI will have updated docs that treat it like any other valid install option instead of just an older blog post that seems like it doesn't get the attention other routes do. Even if it is well supported, it still makes it feel more like a hack than something I'm intended to do.
